backend/canisters/user/impl/src/updates/create_community.rs

```diff
@@ -111,7 +112,17 @@ fn prepare(args: Args, state: &RuntimeState) -> Result<PrepareResult, Response>
 }
 
 fn default_channels_valid(default_channels: &Vec<String>) -> bool {
-    !default_channels.is_empty() && default_channels.iter().all(|channel| validate_name(channel, true).is_ok())
+    if default_channels.is_empty() || default_channels.iter().any(|channel| validate_name(channel, true).is_err()) {
+        return false;
+    }
+
+    let names: HashSet<String> = default_channels.iter().map(|name| name.to_lowercase()).collect();
+
+    if names.len() != default_channels.len() {
```

Comment:
`names.len() == default_channels.len()`

No need for return

Reply:
I was giving room for further future clauses...

Reply:
```
if clause1 {
    return false;
}

if clause2 {
    return false;
}

true
```